scipy/stats/_morestats.py

```diff
@@ -4047,12 +4058,16 @@ def directionalmean(samples, *, axis=0, normalize=True):
 
     Returns
     -------
-    directionalmean : ndarray
+    mean_resultant : ndarray
```

Comment:
This is not the "mean resultant", though. The "mean resultant" does not always have unit length, but this attribute is always normalized. This quantity is called the "mean direction" [here](https://github.com/scipy/scipy/pull/16785#discussion_r945166683), but I think "mean resultant direction" would also be reasonable.

Since this is close to merging, I'll give it one last shot.

In one sense, I'm glad that we've come around to including `mean_resultant` in the names of the attributes, because "mean resultant" is the standard name associated with these quantities in the literature. With this in mind, though, I'm surprised that we are against just calling the function `mean_resultant` and giving it attributes `direction` and `length`.

I agree with the discoverability argument: we want functions to be discoverable by those who aren't experts. But I don't think that users actually look for functions by scanning the list of function names manually and missing all the ones that don't have the name they expect. If they're like me, they search either with a browser's "find text" feature or using a search engine. If we indicate in the one-line function description that the `mean_resultant` function calculates measures of "preferred (average) direction and dispersion", there's little chance that users will miss this function because of its name, considering that it's at the top of the "Directional Statistics" section.

If we have concrete plans for what other attributes/methods we might like to add to the return object of a `directional_stats` function, I'm OK calling the function `directional_stats` and giving it attributes `mean_direction` and `mean_resultant_length` for now. If we don't think it's likely this will be extended, this is one last call for a `mean_resultant` function with attributes `direction` and `length`.

Reply:
True, `mean_resultant` is not the right term here.

I would opt for `directional_stats` with output attributes `mean_direction`/`mean_resultant_length` then.

About discoverability: what I would love to do is just use autocompletion in an IDE to find a function that is probably a good fit for what I would do. For SciPy, that is often difficult even with modules I have used frequently. For the functionality we are implementing here, I would type in sth like `stats.dir` and see what comes up.

Reply:
@rkern you've commented on related issues. Can you help us choose between `mean_resultant` and `directional_stats` as the new name of this function?

Reply:
:+1:  for `directional_stats`

Reply:
Thanks @rkern @mdhaber @dschmitz89! I have changed `mean_resultant` to `mean_direction` and kept `directional_stats`.